libs/clients/smartsolutions/src/lib/smartSolutions.api.ts

```diff
@@ -84,75 +138,42 @@ export class SmartSolutionsApi {
         }
       }
     `
-    let res: Response | null = null
 
     const graphql = JSON.stringify({
       query: listPassesQuery,
       variables: {},
     })
 
-    try {
-      res = await this.fetchUrl(graphql)
-    } catch (e) {
-      this.logger.warn('Unable to retrieve pk passes', {
-        exception: e,
-        category: LOG_CATEGORY,
-      })
-    }
-
-    if (!res) {
-      this.logger.warn('Unable to get pk passes, null from fetch', {
-        category: LOG_CATEGORY,
-      })
-      return null
-    }
+    const response = await this.fetchData<
+      ServiceResponse<{ passes?: { data: Array<Pass> } }>
+    >(graphql)
 
-    if (!res.ok) {
-      const responseErrors: PkPassServiceErrorResponse = {}
-      try {
-        const json = await res.json()
-        responseErrors.message = json?.message ?? undefined
-        responseErrors.status = json?.status ?? undefined
-        responseErrors.data = json?.data ?? undefined
-      } catch {
-        // noop
+    if (response.data) {
+      return {
+        ok: true,
+        data: response.data.data?.passes?.data ?? [],
       }
-
-      this.logger.warn('Expected 200 status for list pkpasses', {
-        status: res.status,
-        statusText: res.statusText,
-        category: LOG_CATEGORY,
-        ...responseErrors,
-      })
-      return null
     }
 
-    let json: unknown
-    try {
-      json = await res.json()
-    } catch (e) {
-      this.logger.warn('Unable to parse JSON for pk passes', {
-        exception: e,
-        category: LOG_CATEGORY,
-      })
-      //return null
-    }
-
-    const response = json as ListPassesResponse
-
-    if (response?.data?.passes?.data) {
-      const passesDTO: ListPassesDTO = {
-        data: response.data.passes.data,
+    if (response.error) {
+      return {
+        ok: false,
+        error: response.error,
       }
-      return passesDTO
     }
 
-    return null
+    return {
+      ok: false,
+      error: {
+        serviceCode: 99,
+        message: 'Unknown error',
+      },
+    }
   }
 
   async verifyPkPass(
     payload: DynamicBarcodeDataInput,
-  ): Promise<VerifyPassResult | null> {
+  ): Promise<Result<Pass & { valid: boolean }>> {
```

Comment:
It might be more clear to split this into it's own type, e.g.
`type PassWithValidation = Pass & { valid: boolean }`